File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/StudentJobLinkResponseDto.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.LinkType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudentJobLinkResponseDto {
    
    private Long id;
    private Long studentId;
    private String studentName;
    private Long jobId;
    private String jobTitle;
    private String jobCompany;
    private LinkType type;
    private LocalDateTime createdAt;
} 

File: OrienGO-Frontend/src/app/model/dto/StudentJobLinkResponse.dto.ts
// src/app/model/dto/student-job-link-response.dto.ts

import { LinkType } from "../enum/LinkType.enum";

export interface StudentJobLinkResponseDto {
  id: number;
  studentId: number;
  studentName?: string;
  jobId: number;
  jobTitle?: string;
  jobCompany?: string;
  type: LinkType;
  createdAt?: string; // ISO date string from backend
}
